### PR TITLE
Flexible array member is not supported in C++11, so emit an array of …

### PR DESCRIPTION
**Pull request by syoyo**
…size 1.

### CODE REVIEW

File: spirv_cpp.cpp [hunk 1]
```diff
@@ -36,6 +36,64 @@ void CompilerCPP::emit_buffer_block(const SPIRVariable &var)
...
+	uint64_t memberflags = 0;
+	auto &memb = meta[type.self].members;
+	if (index < memb.size())
+		memberflags = memb[index].decoration_flags;
+
+	return join(layout_for_member(type, index), flags_to_precision_qualifiers_glsl(membertype, memberflags),
+	            type_to_glsl(membertype), " ", to_member_name(type, index), type_to_array_cpp(membertype));
+}
+
+void CompilerCPP::emit_struct(const SPIRType &type)
```
Comment (HansKristian-ARM) [inline spirv_cpp.cpp]: Is there anything C++ specific about this method? It's not marked as override or virtual, so it looks like it can be removed?

**Comment by syoyo** [comment]
The purpose of this PR is just emit `[1] ` when size is zero  in `type_to_array_glsl`, so you can find a way to fix it more efficient way. It may better to introducing some flag args  in `type_to_array_glsl`. This will  reuse CompilerGLSL code as much as possible.

**Comment by HansKristian-ARM** [comment]
It should be easier to add a flag to CompilerGLSL::BackendVariations. That struct is designed to handle tiny variations like this without having to reimplement virtuals.

**Comment by HansKristian-ARM** [comment]
Implemented it with a BackendVariations check instead.

**Comment by syoyo** [comment]
Cool! Thanks!

**Pull request closed without merging**